lighthouse-core/gather/gatherers/dobetterweb/geolocation-on-start.js

```diff
@@ -37,12 +37,20 @@ class GeolocationOnStart extends Gatherer {
    * @return {!Promise<!Array<{url: string, line: number, col: number}>>}
    */
   afterPass(options) {
-    return options.driver.queryPermissionState('geolocation')
-      .then(state => {
-        if (state === 'granted' || state === 'denied') {
-          throw new Error('Unable to determine if this permission was requested on page load ' +
-              'because it had already been set. Try resetting the permission and running ' +
-              'Lighthouse again.');
+    return options.driver.evaluateAsync('(function(){return window.isSecureContext;})()')
```

Comment:
Note: `isSecureContext` returns true for `https://` and `http://localhost`. The latter is a secure context for testing purposes. We may want to consider moving the HTTPs gatherer to this or supplementing it with this check for https://github.com/GoogleChrome/lighthouse/issues/1175.